### PR TITLE
[fix][broker]avoid the bestBroker choice broker exceed candidates when selectBroker in LeastResourceUsageWithWeight

### PR DESCRIPTION
**Pull request by Nicklee007**
### Motivation

Avoid the bestBroker choice broker exceed candidates when selectBroker in LeastResourceUsageWithWeight.
When loadManager leader running a period of time,   `brokerAvgResourceUsageWithWeight` will contain almost all brokers, the judge will cause we selected broker exceed candidates, it's out of expectation.

### Modifications

1. check if `candidates` contain the broker before add the broker to `bestBrokers`.

### Documentation

- [X] `doc-not-needed` 
(Please explain why)

### CODE REVIEW

**Review by shibd — CHANGES_REQUESTED** [review]
Nice catch! Could you add a unit test to protect this change?

**Review by gaozhangmin — APPROVED** [review]

**Comment by codelipenghui** [comment]
> Nice catch! Could you add a unit test to protect this change?

+1 Please add test to avoid regression.

**Comment by HQebupt** [comment]
Nice catch. +1

**Review by HQebupt — APPROVED** [review]

**Comment by Nicklee007** [comment]
> Nice catch! Could you add a unit test to protect this change?

@codelipenghui @shibd add unit test,PTAL~

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LeastResourceUsageWithWeight.java [hunk 1]
```diff
@@ -139,7 +139,8 @@ public Optional<String> selectBroker(Set<String> candidates, BundleData bundleTo
         final double diffThreshold =
                 conf.getLoadBalancerAverageResourceUsageDifferenceThresholdPercentage() / 100.0;
         brokerAvgResourceUsageWithWeight.forEach((broker, avgResUsage) -> {
-            if (avgResUsage + diffThreshold <= avgUsage) {
+            if ((avgResUsage + diffThreshold <= avgUsage)
+                    && candidates.contains(broker)) {
                 bestBrokers.add(broker);
             }
         });
```
Comment (Demogorgon314) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LeastResourceUsageWithWeight.java:146]: We can change it to be more simple, right?
```java
for (String broker : candidates) {
    Double avgResUsage = brokerAvgResourceUsageWithWeight.getOrDefault(broker, Double.MAX_VALUE);
    if ((avgResUsage + diffThreshold <= avgUsage)) {
        bestBrokers.add(broker);
    }
}
```

**Review by Demogorgon314 — APPROVED** [review]
Nice catch! LGTM.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LeastResourceUsageWithWeight.java (hunk 1 above)
Comment (Nicklee007) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LeastResourceUsageWithWeight.java]: you're right, `candidates` is smaller than `brokerAvgResourceUsageWithWeight`, changed it.

**Review by shibd — APPROVED** [review]

**Comment by codelipenghui** [comment]
Removed the release/2.10.2 and release/2.9.4 labels since we don't have LeastResourceUsageWithWeight on branch-2.10 and branch-2.9

**Pull request merged**